lib/handler/configurator/proxy.c

```diff
@@ -142,6 +152,77 @@ static int on_config_ssl_cafile(h2o_configurator_command_t *cmd, h2o_configurato
     return ret;
 }
 
+static h2o_cache_t *create_ssl_session_cache(size_t capacity, unsigned lifetime)
+{
+    return h2o_cache_create(H2O_CACHE_FLAG_MULTITHREADED, capacity, (uint64_t)lifetime * 1000, h2o_socket_ssl_destroy_session_cache_entry);
+}
+
+static int on_config_ssl_session_cache(h2o_configurator_command_t *cmd, h2o_configurator_context_t *ctx, yoml_t *node)
+{
+    struct proxy_configurator_t *self = (void *)cmd->configurator;
+    size_t i;
+    size_t capacity = self->vars->ssl_session_cache.capacity;
+    unsigned lifetime = self->vars->ssl_session_cache.lifetime;
+    if (capacity == 0 || lifetime == 0) {
+        capacity = H2O_DEFAULT_PROXY_SSL_SESSION_CACHE_CAPACITY;
+        lifetime = H2O_DEFAULT_PROXY_SSL_SESSION_CACHE_LIFETIME;
```

Comment:
Throughout the PR, either `capacity` or `lifetime` being zero is used as an indication of the cache being disabled.

The approach seems unnecessary complex to me.

How about using specifying either of the value being zero as a mark for indicating a disabled cache?

Also, it might also be a good idea to emit an error in case either of the two is set to zero but the other is not to avoid confusion, instead of silently disabling the cache.